drivers/LinstorSR.py

```diff
@@ -1907,7 +1907,30 @@ def detach(self, sr_uuid, vdi_uuid) -> None:
             vdi_uuid = parent_vdi_uuid
 
     @override
-    def resize(self, sr_uuid, vdi_uuid, size) -> str:
+    def resize(self, sr_uuid: str, vdi_uuid: str, size: int) -> str:
+        def compute_vdi_sizes(vdi: LinstorVDI) -> Tuple[int, int]:
+            if vdi.vdi_type == vhdutil.VDI_TYPE_RAW:
+                return vdi.size, LinstorVolumeManager.round_up_volume_size(size)
+
+            old_volume_size = vdi.utilisation
+            if vdi.sr._provisioning == 'thin':
+                # VDI is currently deflated, so keep it deflated.
+                return old_volume_size, old_volume_size
+
+            return old_volume_size, LinstorVhdUtil.compute_volume_size(size, vdi.vdi_type)
+
+        def resize_vdi(vdi: LinstorVDI, old_volume_size: int, new_volume_size) -> None:
+            if vdi.vdi_type == vhdutil.VDI_TYPE_RAW:
+                vdi._linstor.resize_volume(vdi.uuid, new_volume_size)
+                return
+
+            if new_volume_size != old_volume_size:
+                vdi.sr._vhdutil.inflate(
+                    vdi.sr._journaler, vdi.uuid, vdi.path,
+                    new_volume_size, old_volume_size
+                )
+            vdi.sr._vhdutil.set_size_virt_fast(vdi.path, size)
+
```

Comment:
Not really against it but the refactor seems a bit out of scope here.

Reply:
I've put this refactor in it's dedicated commit